src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/CodeGeneration/CSharpCodeGenerationContextInfo.cs

```diff
@@ -21,6 +22,9 @@ public CSharpCodeGenerationContextInfo(CodeGenerationContext context, CSharpCode
         public new CSharpCodeGenerationOptions Options { get; }
         public new CSharpCodeGenerationService Service { get; }
 
+        protected override SyntaxGenerator GeneratorImpl
+            => Service.LanguageServices.GetRequiredService<SyntaxGenerator>();
+
```

Comment:
Does this need to be in language specific layer? Can it be moved down to `CodeGenerationContextInfo` type?

Reply:
I'll try that in follow-up!